nomad/job_endpoint_hook_expose_check.go

```diff
@@ -0,0 +1,228 @@
+package nomad
+
+import (
+	"strconv"
+	"strings"
+
+	"github.com/hashicorp/nomad/nomad/structs"
+	"github.com/pkg/errors"
+)
+
+type jobExposeCheckHook struct{}
+
+func (jobExposeCheckHook) Name() string {
+	return "expose-check"
+}
+
+// Mutate will scan every task group for group-services which have checks defined
+// that have the Expose field configured, and generate expose path configurations
+// extrapolated from those check definitions.
+func (jobExposeCheckHook) Mutate(job *structs.Job) (_ *structs.Job, warnings []error, err error) {
+	for _, tg := range job.TaskGroups {
+		for _, s := range tg.Services {
+			for _, c := range s.Checks {
+				if c.Expose {
```

Comment:
Maybe switch this to `if !c.Expose { continue }` to save at least one level of indentation for readability.

Reply:
I've had to do this a few times and groan each time. What do you folks think about adding helper funcs where fitting to make this a bit more readable. Something like:
```
job.WalkTGServiceChecks(func(tg *structs.TaskGroup, s *struct.Service, c *struct.ServiceCheck){
	if !c.Expose { return }
	...
})
```

Reply:
I was having similar thoughts about having a walk func - if we go that route we'd probably want to enable some err control similar to what `filepath.Walk` [provides](https://golang.org/src/path/filepath/path.go?#L369)